modules/imgproc/test/test_houghlines.cpp

```diff
@@ -340,6 +340,53 @@ TEST(HoughLines, regression_21983)
     EXPECT_NEAR(lines[0][1], 1.57179642, 1e-4);
 }
 
+TEST(WeightedHoughLines, horizontal)
+{
+    Mat img(25, 25, CV_8UC1, Scalar(0));
+    // draw lines. from top to bottom, stronger to weaker.
+    line(img, Point(0, 6), Point(25, 6), Scalar(255));
+    line(img, Point(0, 12), Point(25, 12), Scalar(254));
+    line(img, Point(0, 18), Point(25, 18), Scalar(253));
+    
+    // detect lines
+    std::vector<Vec2f> lines;
+    int threshold{253*25-1};
+    bool use_edgeval{true};
+    HoughLines(img, lines, 1, CV_PI/180, threshold, 0, 0, 0.0, CV_PI, use_edgeval);
```

Comment:
BTW, test doesn't look representative.

```
    int threshold{25-1};
    HoughLines(img, lines, 1, CV_PI/180, threshold, 0, 0, 0.0, CV_PI);
```

works well without this patch.

if HoughLines() input comes from Canny() then this input is in set of `[0, 255]` without intermediate values.

How is intermediate values could be prepared on edges?

Reply:
Oh, I'm very sorry!  I didn't noticed this message.
I will take a look the above.

Reply:
Just a moment please for the fixation.
I'm still struggling to build latest 4.x branch.
I reported my error [here](https://github.com/opencv/opencv/issues/26620)

Reply:
Now I fixed the unit test. and confirmed it.
If I set " use_edgeval{false}" for both TEST(WeightedHoughLines, horizontal), and TEST(WeightedHoughLines, diagonal), and run the command;
```bash
$ ./ext/sensyn-opencv/bin/opencv_test_imgproc --gtest_filter=WeightedHoughLines.*
```
, it fails as below;
```test_log
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from WeightedHoughLines
[ RUN      ] WeightedHoughLines.horizontal
/home/mas/proj/sensyn/sensyn-vision/ext/sensyn-opencv/modules/imgproc/test/test_houghlines.cpp:358: Failure
Expected equality of these values:
  3
  lines.size()
    Which is: 0
[  FAILED  ] WeightedHoughLines.horizontal (1 ms)
[ RUN      ] WeightedHoughLines.diagonal
/home/mas/proj/sensyn/sensyn-vision/ext/sensyn-opencv/modules/imgproc/test/test_houghlines.cpp:382: Failure
Expected equality of these values:
  2
  lines.size()
    Which is: 0
[  FAILED  ] WeightedHoughLines.diagonal (0 ms)
[----------] 2 tests from WeightedHoughLines (1 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (1 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] WeightedHoughLines.horizontal
[  FAILED  ] WeightedHoughLines.diagonal

 2 FAILED TESTS
```

But when I revised to "use_edgeval{true}" again as the pushed code, the result become success as below.
```test_log
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from WeightedHoughLines
[ RUN      ] WeightedHoughLines.horizontal
[       OK ] WeightedHoughLines.horizontal (1 ms)
[ RUN      ] WeightedHoughLines.diagonal
[       OK ] WeightedHoughLines.diagonal (0 ms)
[----------] 2 tests from WeightedHoughLines (1 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (1 ms total)
[  PASSED  ] 2 tests.
```

Reply:
@opencv-alalek   I'm very sorry for my very late reply, but could you kindly review the code again?